core/src/main/java/com/linecorp/armeria/common/HttpHeadersBuilder.java

```diff
@@ -0,0 +1,509 @@
+     *
+     * @param name the name of the header to search
+     * @param defaultValue the default value
+     * @return the {@code double} value of the first value in insertion order or {@code defaultValue} if there
+     *         is no such value or it can't be converted to {@code double}.
+     */
+    double getDoubleAndRemove(CharSequence name, double defaultValue);
+
+    /**
+     * Removes all the headers with the specified name and returns the header value which was added first.
+     *
+     * @param name the name of the header to retrieve
+     * @return the milliseconds value of the first value in insertion order or {@code null} if there is no such
+     *         value or it can't be converted to milliseconds.
+     */
+    @Nullable
+    Long getTimeMillisAndRemove(CharSequence name);
+
+    /**
+     * Removes all the headers with the specified name and returns the header value which was added first.
+     *
+     * @param name the name of the header to retrieve
+     * @param defaultValue the default value
+     * @return the milliseconds value of the first value in insertion order or {@code defaultValue} if there is
+     *         no such value or it can't be converted to milliseconds.
+     */
+    long getTimeMillisAndRemove(CharSequence name, long defaultValue);
+
+    /**
+     * Adds a new header with the specified {@code name} and {@code value}.
+     *
+     * @param name the header name
+     * @param value the header value
+     * @return {@code this}
+     */
+    HttpHeadersBuilder add(CharSequence name, String value);
+
+    /**
+     * Adds new headers with the specified {@code name} and {@code values}. This method is semantically
+     * equivalent to
+     * <pre>{@code
+     * for (String value : values) {
+     *     headers.add(name, value);
+     * }
+     * }</pre>
+     *
+     * @param name the header name
+     * @param values the header values
+     * @return {@code this}
+     */
+    HttpHeadersBuilder add(CharSequence name, Iterable<String> values);
+
+    /**
+     * Adds new headers with the specified {@code name} and {@code values}. This method is semantically
+     * equivalent to
+     * <pre>{@code
+     * for (String value : values) {
+     *     headers.add(name, value);
+     * }
+     * }</pre>
+     *
+     * @param name the header name
+     * @param values the header values
+     * @return {@code this}
+     */
+    HttpHeadersBuilder add(CharSequence name, String... values);
+
+    /**
+     * Adds all header names and values of the specified {@code headers}.
+     *
+     * @return {@code this}
+     * @throws IllegalArgumentException if {@code headers == this}.
+     */
+    HttpHeadersBuilder add(
+            Iterable<? extends Entry<? extends CharSequence, String>> headers);
+
+    /**
+     * Adds a new header. Before the {@code value} is added, it's converted to a {@link String}.
```

Comment:
Can you link to the string conversion logic? I think it can help users understand what will happen

Reply:
By the way, I think this can just be `@see StringValueConverter` - lucky that's in `common` :D

https://github.com/line/armeria/pull/1731/files#diff-71e30b4cf4ff10735fc468087527be95